### PR TITLE
Vision foundation model

### PR DESCRIPTION
**Pull request by rjaincc**
# Description
Merging Image Foundation Model examples to main. 
Runs captured [here](https://loop.microsoft.com/p/eyJ3Ijp7InUiOiJodHRwczovL21pY3Jvc29mdC5zaGFyZXBvaW50LmNvbS8%2FbmF2PWN6MGxNa1ltWkQxaUlWOUlPV3RUV0RKelVrVkhURkZVUjAxeWJFTTRXVFpqVjBkNGMySkRhWEJDY25WaFUyVkxWRlJaTTAxWllVWnJWemxqUVhkU05qWkhkMVJZV2tzdGQyMG1aajB3TVRKU1VrZEdSVXhDUmtoU1JWSktSek15UmtkS1VrVkRVVUUwUzBNMk5GSkpKbU05Sm1ac2RXbGtQVEUlM0QiLCJyIjpmYWxzZX0sInAiOnsidSI6Imh0dHBzOi8vbWljcm9zb2Z0LnNoYXJlcG9pbnQuY29tLzpmbDovci9zaXRlcy80Yzg4OGNjMC1iMzZkLTRlMGQtOTRhZS00N2MzMGJkY2IwMTMvU2hhcmVkJTIwRG9jdW1lbnRzL0xvb3BBcHBEYXRhL1VudGl0bGVkJTIwcGFnZSUyMDEubG9vcD9kPXc4YTYwMjllYTgxNDg0YzM5OGM1ZWUxOGJmYjgzNzIyNiZjc2Y9MSZ3ZWI9MSZuYXY9Y3owbE1rWnphWFJsY3lVeVJqUmpPRGc0WTJNd0xXSXpObVF0TkdVd1pDMDVOR0ZsTFRRM1l6TXdZbVJqWWpBeE15WmtQV0loWDBnNWExTllNbk5TUlVkTVVWUkhUWEpzUXpoWk5tTlhSM2h6WWtOcGNFSnlkV0ZUWlV0VVZGa3pUVmxoUm10WE9XTkJkMUkyTmtkM1ZGaGFTeTEzYlNabVBUQXhNbEpTUjBaRlVFdEdSbEZKVlZORlFraEdSMGxaV0ZoQ1VsQTFXVWMwVWtjbVl6MGxNa1ltWm14MWFXUTlNU1poUFV4dmIzQkJjSEFtY0QwbE5EQm1iSFZwWkhnbE1rWnNiMjl3TFhCaFoyVXRZMjl1ZEdGcGJtVnlKbmc5SlRkQ0pUSXlkeVV5TWlVelFTVXlNbFF3VWxSVlNIaDBZVmRPZVdJelRuWmFibEYxWXpKb2FHTnRWbmRpTW14MVpFTTFhbUl5TVRoWmFVWm1VMFJzY2xVeFozbGpNVXBHVWpCNFVsWkZaRTVqYlhoRVQwWnJNbGt4WkVobFNFNXBVVEpzZDFGdVNqRlpWazVzVXpGU1ZWZFVUazVYVjBaSFlURmpOVmt3UmpOVmFsa3lVak5rVlZkR2NFeE1XR1IwWmtSQmVFMXNTbE5TTUZwR1ZFVktSMU5HU2taVmEzQklUWHBLUjFJd2NGTlNWVTVTVVZSU1RGRjZXVEJWYTJzbE0wUWxNaklsTWtNbE1qSnBKVEl5SlROQkpUSXlZVGhrTURrMk5ERXRPVEpoTnkwME5XVmpMV0l6WkRRdFlUUXdOR1psWmpFMllUVTBKVEl5SlRkRSIsInIiOmZhbHNlfSwiaSI6eyJpIjoiYThkMDk2NDEtOTJhNy00NWVjLWIzZDQtYTQwNGZlZjE2YTU0In19).

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    Pytorch Model registered in AzureML Asset.
+
+    The continual finetune flag will be set to true in this case. If you want to resume from previous training state, set *resume_from_checkpoint* flag to True in [finetune component](mmd_finetune_component.md/#39-resume-from-checkpoint).
+
+2. _mlflow_model_ (mlflow_model, optional)
+
+    MlFlow Model registered in AzureML Asset. Some MMDetection models are registered in azureml-preview registry and can be used directly. The user can also register MMDetection models into their workspace or organisation's registry, and use them.
+
+    Following models are registered in azureml-preview registry, and can be used directly.
```
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md:18]: it should be azureml right?

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_model_import_component.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    Pytorch Model registered in AzureML Asset.
+
+    The continual finetune flag will be set to true in this case. If you want to resume from previous training state, set *resume_from_checkpoint* flag to True in [finetune component](transformers_finetune_component.md/#38-resume-from-checkpoint).
+
+2. _mlflow_model_ (mlflow_model, optional)
+
+    MlFlow Model registered in AzureML Asset. Some HuggingFace models are registered in azureml-preview registry and can be used directly. The user can also register HuggingFace models into their workspace or organisation's registry, and use them.
+
+    Following models are registered in azureml-preview registry, and can be used directly.
```
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_model_import_component.md:19]: same, please check.

**Review by gauravrajguru — CHANGES_REQUESTED** [review]
registry name

**Review by gauravrajguru — APPROVED** [review]

**Comment by skasturi** [comment]
Thanks for updating the code quality WF. 

Could you please also update the import model gate to include the updates?

**Review by nvijayrania — APPROVED** [review]

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
+# MMDetection Model Finetune Component
+This component enables finetuning of pretrained models on custom or pre-available datasets. The component supports Deepspeed and ONNXRuntime configurations for performance enhancement. 
+The components can be seen here ![as shown in the figure](../../images/image_mmd_od_is_finetune_components.png)
```
Comment (nvijayrania) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:3]: Nit: we can have link to the actual component in azureml registry.

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+# 3. Parameters
+1. _model_family_ (string, required)
+
+    Which framework the model belongs to.
+    It could be one of [`MmDetectionImage`]
+
+2. _model_name_ (string, optional)
+
+    Please select models from AzureML Model Assets for all supported models. For MMDetection models, which are not supported in AuzreML model registry, the model's config name is required, same as it's specified in MMDetection Model Zoo. For e.g. fast_rcnn_r101_fpn_1x_coco for [this config file](https://github.com/open-mmlab/mmdetection/blob/master/configs/fast_rcnn/fast_rcnn_r101_fpn_1x_coco.py). You can see the comprehensive list of model configs [here] (https://github.com/open-mmlab/mmdetection/tree/master/configs) and the documentation of model zoo [here] (https://github.com/open-mmlab/mmdetection/blob/main/docs/en/model_zoo.md).
```
Comment (nvijayrania) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md:57]: typo: AuzreML  --> AzureML

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_model_import_component.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+# 3. Parameters
+1. _model_family_ (string, required)
+
+    Which framework the model belongs to.
+    It could be one of [`HuggingFaceImage`]
+
+2. _model_name_ (string, optional)
+
+    Please select models from AzureML Model Assets for all supported models.
+    For HuggingFace models, which are not supported in AuzreML model registry, input HuggingFace model name here. You can see supported image-classification models [here](https://huggingface.co/models?pipeline_tag=image-classification&library=transformers).
```
Comment (nvijayrania) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_model_import_component.md:55]: typo: AuzreML  --> AzureML

**Review by skasturi — APPROVED** [review]

File: .github/workflows/code-quality.yml [hunk 6]
```diff
@@ -18,4 +18,4 @@ jobs:
       - name: Check Pull Request Size
         run: |
           git fetch origin ${{ github.event.pull_request.base.ref }} --quiet # Need to manually fetch base branch in CI
-          python ./.github/scripts/commit-filesize-diff-summary.py --limit 2MB origin/${{ github.event.pull_request.base.ref }}..HEAD
+          python ./.github/scripts/commit-filesize-diff-summary.py --limit 4MB origin/${{ github.event.pull_request.base.ref }}..HEAD
```
Comment (skasturi) [inline .github/workflows/code-quality.yml:21]: While this will unblock this PR, let us revert it once this PR gets merged.

**Review by RSavage2 — APPROVED** [review]

File: .github/workflows/code-quality.yml (hunk 6 above)
Comment (rjaincc) [reply .github/workflows/code-quality.yml:21]: Will raise the PR right after.

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+
+34. _save_steps_ (int, optional)
+
+    Number of updates steps before two checkpoint saves if save_strategy="steps".
+    The default value is 500.
+    Please note that the saving steps should be a multiple of the evaluation steps.
+
+35. _save_total_limit_ (int, optional)
+
+    If a value is passed, will limit the total amount of checkpoints. Deletes the older checkpoints in output_dir. If the value is -1 saves all checkpoints". The default value is -1.
```
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:195]: nit: total amount of -> total number of

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+
+36. _early_stopping_ (bool, optional)
+
+    If set to true, early stopping is enabled. The default value is false.
+
+37. _early_stopping_patience_ (int, optional)
+
+    Stop training when the specified metric worsens for early_stopping_patience evaluation calls. The default value is 1.
```
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:203]: worsens -> doesn't improve

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+
+40. _save_as_mlflow_model_ (bool, optional)
+
+    Save as mlflow model with pyfunc as flavour. The default value is true.
+
+
+# 2. Outputs
+1. _output_dir_pytorch_ (custom_model, required)
+
+    The folder containing finetuned model output with checkpoints, model config, optimzer and scheduler states and random number states in case of distributed training.
```
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:223]: optimzer -> optimizer

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+
+2. _output_dir_mlflow_ (URI_FOLDER, optional)
+
+    Output dir to save the finetuned model as mlflow model.
```
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:227]: Output dir -> Output directory

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+
+
+# 4. Run Settings
+
+This setting helps to choose the compute for running the component code. **For the purpose of finetune, gpu compute should be used**. We recommend using Standard_NC6s or Standard_NC6s_v3 compute.
```
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:232]: nit: links to the compute types?

**Review by vadthyavath — CHANGES_REQUESTED** [review]

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    | [yolof_r50_c5_8x8_1x_coco](https://ml.azure.com/registries/azureml/models/yolof_r50_c5_8x8_1x_coco/version/3) | azureml registry |
+    | [mask_rcnn_swin-t-p4-w7_fpn_1x_coco](https://ml.azure.com/registries/azureml/models/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/version/3) | azureml registry |
+
+    Below is the folder structure of a registered MLFlow model.
+
+    ![Mlflow Model Tree](../../images/mmd_mlflow_model.png)
+
+    - In the finetuned model, all the augmentations should be specified in artifacts/augmentations.yaml file.
+    - All the model files should be stored in artifacts/<model_name>*.pth or artifacts/<model_name>*.py file.
+    - **`MLmodel`** is a yaml file and this should contain relavant information. See the sample MLmodel file [here](../../sample_files/MMDMLmodel). Please note that the
```
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md:33]: this link is not working

**Pull request merged**